src/nvme/json.c

```diff
@@ -193,16 +193,20 @@ static struct json_object *parse_json(nvme_root_t r, int fd)
 {
 	char buf[JSON_FILE_BUF_SIZE];
 	struct json_object *obj = NULL;
-	struct printbuf *pb;
+	char *str;
```

Comment:
The compiler seems not be able to figure out that str is always defined when calling `json_tokener_parse_ex()`. So you might just want to explicitly initialize it here to NULL.

Reply:
Thanks for your review comment. Just updated the patch to add allocated data length checking instead of NULL pointer initialization.

Reply:
Sorry refixed the patch to initialize the pointer to NULL as mentioned.

Reply:
No worries, that's why we have some CI builds :)